### PR TITLE
[Link] Update styles

### LINKED ISSUE #235: [Link] update style

### PR DESCRIPTION
**Pull request by kylegach**
### Description
- Remove fontWeight, fontSize
- Remove includeStyleReset
- Add color_text_variant_state variables
- Change color_link_* to color_text_primary_* theme variables, for better parellelism
- Add variant prop & styles
- Remove rootEl option from Link's createStyledComponent; it wasn't doing anything
- Add correct focus styles
- Update examples
- Update snapshots

Closes #235 

### Motivation and context

The current Link doesn't match spec, and the hover/focus styles were not discernible.

### Screenshots or videos, if appropriate

screen shot 2017-08-10 at 1 28 53 pm

### How to test
1. `cd mineral-ui/packages/link && npm start`
1. Confirm it looks like the spec

### Types of changes

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

Breaking: Some Link theme variables were removed

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered
* [x] Documentation created or updated

### How does this PR make you feel?

Link

### CODE REVIEW

File: packages/link/src/Link.js [hunk 1]
```diff
@@ -29,27 +29,31 @@ type Props = {
 
 const linkStyles = (props, baseTheme) => {
   const theme = {
+    Link_borderColor_focus: baseTheme.borderColor_focus,
     Link_color: baseTheme.color_link,
     Link_color_hover: baseTheme.color_link_hover,
     Link_color_focus: baseTheme.color_link_focus,
     Link_color_active: baseTheme.color_link_active,
-    Link_fontSize: `${2 * baseTheme.fontSize_base}px`,
-    Link_fontWeight: baseTheme.fontWeight_regular,
```
Comment (brentertz) [inline packages/link/src/Link.js]: 👍 Much simpler this way.

File: packages/link/src/Link.js [hunk 2]
```diff
@@ -65,7 +69,6 @@ const linkStyles = (props, baseTheme) => {
 export default function Link({ children, element = 'a', ...restProps }: Props) {
   const Root = createStyledComponent(element, linkStyles, {
     displayName: 'Link',
-    includeStyleReset: true,
```
Comment (brentertz) [inline packages/link/src/Link.js]: 👍 Much simpler this way.

File: packages/link/src/Link.js [hunk 3]
```diff
@@ -29,27 +29,31 @@ type Props = {
...
   };
 
   return {
     color: theme.Link_color,
-    fontWeight: theme.Link_fontWeight,
-    fontSize: theme.Link_fontSize,
-    outline: null, // Using the browser's focus styles until design is finalized
+    textDecoration: 'none',
 
+    '&:visited': {
```
Comment (brentertz) [inline packages/link/src/Link.js]: Assuming `:visited` is going away based on external discussions...

File: packages/link/src/__demo__/examples/link.js [hunk 4]
```diff
@@ -15,16 +15,41 @@
...
+const Bold = createStyledComponent('p', {
+  fontWeight: 700
+});
+const Serif = createStyledComponent('p', {
+  fontFamily: 'serif'
+});
+const Small = createStyledComponent('p', {
+  fontSize: '0.75em'
+});
+
```
Comment (coldpour) [inline packages/link/src/__demo__/examples/link.js]: I think we should include these definitions in the example code so that it's clear how this works

Comment (kylegach) [inline packages/link/src/__demo__/examples/link.js]: I disagree. Both the example description and live preview make it more than clear what's going on (Links look like whatever they're placed in). The actual code used to demonstrate that isn't important.

File: packages/link/src/Link.js (hunk 3 above)
Comment (kylegach) [inline packages/link/src/Link.js]: Yeah, possibly (hopefully).

**Comment by kylegach** [comment]
Note: Using Link with React Router's Link as `element` will throw a warning about the `variant` prop when in dev mode. I'm looking into how to get around this, but didn't think it needed to block this.

**Pull request merged**